python/lsst/ap/association/assoc_db_sqlite.py

```diff
@@ -0,0 +1,479 @@
+        the config.
+        """
+        pipeBase.Task.__init__(self, **kwargs)
+        self.indexer = IndexerRegistry[self.config.indexer.name](
+            self.config.indexer.active)
+        self._db_connection = sqlite3.connect(self.config.db_name)
+        self._db_cursor = self._db_connection.cursor()
+
+    def run(self):
+        """ Not implemented in this subtask. Please use the store and load
+        methods.
+        """
+        Raise(NotImplementedError)
+
+    def commit(self):
+        """ Save changes to the sqlite database.
+        """
+        self._db_connection.commit()
+
+    def close(self):
+        """ Close the connection to the sqlite database.
+        """
+        self._db_connection.close()
+
+    def create_tables(self):
+        """ If no sqlite database exists with the correct tables we can create
+        one using this method.
+
+        Returns
+        -------
+        bool
+            Successfully created a new database with specified tables.
+        """
+
+        obj_schema = make_minimal_dia_object_schema()
+        src_schema = make_minimal_dia_source_schema()
+
+        self._db_cursor.execute(
+            'select name from sqlite_master where type = "table"')
+        db_tables = self._db_cursor.fetchall()
+
+        if db_tables:
+            return False
+        else:
+            name_type_string = ""
+            for sub_schema in obj_schema:
+                tmp_name = sub_schema.getField().getName()
+                tmp_type = afw_to_db_types[
+                    sub_schema.getField().getTypeString()]
+                if tmp_name == 'id':
+                    tmp_type += " PRIMARY KEY"
+                name_type_string += "%s %s," % (tmp_name, tmp_type)
+            name_type_string = name_type_string[:-1]
+            self._db_cursor.execute(
+                "CREATE TABLE dia_objects (%s)" % name_type_string)
+
+            name_type_string = ""
+            for sub_schema in src_schema:
+                tmp_name = sub_schema.getField().getName()
+                tmp_type = afw_to_db_types[
+                    sub_schema.getField().getTypeString()]
+                if tmp_name == 'id':
+                    tmp_type += " PRIMARY KEY"
+                name_type_string += "%s %s," % (tmp_name, tmp_type)
+            name_type_string = name_type_string[:-1]
+            self._db_cursor.execute(
+                "CREATE TABLE dia_sources (%s)" % name_type_string)
+
+            self._db_cursor.execute(
+                "CREATE TABLE dia_objects_to_dia_soruces "
+                "(src_id INTEGER PRIMARY KEY, obj_id INTEGER)")
+
+            self._db_connection.commit()
+
+        return True
+
+    @pipeBase.timeMethod
+    def load(self, ctr_coord, radius):
+        """ Load all DIAObjects and associated DIASources.
+
+        Parameters
+        ----------
+        ctr_coord : lsst.afw.geom.Coord
+            Center position of on on sky circle to load.
```

Comment:
I think this will be mostly agnostic to the type of coordinate it is fed and I'm happy to change it to SpherePoint. Right now what type of coordinate this will be will depend on what the return of the pixelToSky method of a WCS returns. As long as that coordinate object has an angularSeparation method and getLong/Lat methods, the code should work the same.

Reply:
`Wcs` is being replaced by `SkyWcs` , which will return `SpherePoint`. The [names](https://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/classlsst_1_1afw_1_1geom_1_1_sphere_point.html) are a bit different: `separation` instead of `angularSeparation`.